server/src/main/java/org/opensearch/common/time/DateUtils.java

```diff
@@ -272,6 +272,27 @@ public static Instant clampToNanosRange(Instant instant) {
         return instant;
     }
 
+    /**
+     * Clamps the given {@link Instant} to the valid epoch millisecond range.
+     *
+     * - If the input is before {@code Long.MIN_VALUE}, it returns {@code Instant.ofEpochMilli(Long.MIN_VALUE)}.
+     * - If the input is after {@code Long.MAX_VALUE}, it returns {@code Instant.ofEpochMilli(Long.MAX_VALUE)}.
+     * - Otherwise, it returns the input as-is.
+     *
+     * @param instant the {@link Instant} to clamp
+     * @return the clamped {@link Instant}
+     * @throws NullPointerException if the input is {@code null}
+     */
+    public static Instant clampToMillisRange(Instant instant) {
+        if (instant.isBefore(Instant.ofEpochMilli(Long.MIN_VALUE))) {
```

Comment:
It's probably worth extracting `Instant.ofEpochMilli(Long.MIN_VALUE)` and `Instant.ofEpochMilli(Long.MAX_VALUE)` as private constants to avoid creating the objects on every call.

Reply:
Thanks @andrross following the existing static variable declarations I have updated in my latest commit, please check.